zrml/prediction-markets/src/lib.rs

```diff
@@ -2423,11 +2548,44 @@ mod pallet {
             // If the oracle reported right, return the OracleBond, otherwise slash it to
             // pay the correct reporters.
             let mut overall_imbalance = NegativeImbalanceOf::<T>::zero();
-            if report.by == market.oracle && report.outcome == resolved_outcome {
-                Self::unreserve_oracle_bond(market_id)?;
+
+            let report_by_oracle = report.by == market.oracle;
+            let is_correct = report.outcome == resolved_outcome;
+
+            let unreserve_outsider = || -> DispatchResult {
+                if Self::is_outsider_bond_pending(market_id, market, true) {
+                    Self::unreserve_outsider_bond(market_id)?;
+                }
+                Ok(())
+            };
+
+            let slash_outsider = || -> Result<NegativeImbalanceOf<T>, DispatchError> {
+                if Self::is_outsider_bond_pending(market_id, market, true) {
+                    let imbalance = Self::slash_outsider_bond(market_id, None)?;
```

Comment:
I see your point. This is essentially also my concern about resolutions. There are many places inside `on_resolution` where the code potentially could throw an error. That's why we used the approach of `is_outsider_bond_pending` here and are logging the warning. Whenever we add some error case to `slash_outsider_bond` it is not covered by `is_outsider_bond_pending` anymore. But with the current solution there is no way `slash_outsider_bond` could fail, because the assumptions are ensured in the `if` statement above.

[This](https://github.com/zeitgeistpm/zeitgeist/blob/a5f184a12ae0c8850b0d8b907d7800651f0c077d/zrml/prediction-markets/src/lib.rs#L147) is covered by the mentioned `if` statement and the [market existence](https://github.com/zeitgeistpm/zeitgeist/blob/a5f184a12ae0c8850b0d8b907d7800651f0c077d/zrml/prediction-markets/src/lib.rs#L193) is covered by the [`resolution_manager` here](https://github.com/zeitgeistpm/zeitgeist/blob/a5f184a12ae0c8850b0d8b907d7800651f0c077d/zrml/prediction-markets/src/lib.rs#L2896).